library/Fission/CLI/DNS.hs

```diff
@@ -0,0 +1,52 @@
+-- | Update DNS via the CLI
+module Fission.CLI.DNS (update) where
+
+import RIO
+
+import Data.Has
+
+import Servant
+import Servant.Client
+
+import qualified Fission.Config as Config
+import           Fission.Internal.Constraint
+
+import           Fission.IPFS.CID.Types
+
+import qualified Fission.Web.Client      as Client
+import qualified Fission.Web.DNS.Client  as DNS.Client
+
+import           Fission.CLI.Display.Error   as CLI.Error
+import qualified Fission.CLI.Display.Loader  as CLI
+import           Fission.CLI.Display.Success as CLI.Success
+
+import qualified Fission.AWS.Types as AWS
+
+update :: MonadRIO          cfg m
+    => HasLogFunc        cfg
+    => Has Client.Runner cfg
+    => CID
+    -> BasicAuthData
+    -> m (Either ClientError AWS.DomainName)
+update cid@(CID hash) auth = do
+  logDebug $ "Updating DNS to " <> display hash
+
+  Client.Runner runner <- Config.get
+  update' runner auth cid >>= \case
+    Right domain -> do
+      CLI.Success.dnsUpdated $ AWS.getDomainName domain
+      return $ Right domain
```

Comment:
Protip:

```haskell
Right wrapped@(DomainName domain) -> do
  CLI.Success.dnsUpdated domain
  return $ Right wrapped
```

Or with better naming. You can pattern match is what I'm saying 😛